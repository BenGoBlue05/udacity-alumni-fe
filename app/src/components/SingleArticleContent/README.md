## SingleArticleContent Component
A component that ...

### Example

```js
<SingleArticleContent />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
