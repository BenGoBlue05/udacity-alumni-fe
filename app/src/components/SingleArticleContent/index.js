import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Headline from 'grommet-udacity/components/Headline';
import Box from 'grommet-udacity/components/Box';
import Article from 'grommet-udacity/components/Article';
import Image from 'grommet-udacity/components/Image';
import Markdown from 'grommet-udacity/components/Markdown';
import { ArticleCalendar } from 'components';

const SingleArticleContent = ({
  article,
}) => (
  <Article align="center" className={styles.singleArticle}>
    <div className={styles.articleDate}>
      <ArticleCalendar date={article.created_at} />
    </div>
    <Headline align="center">
      {article.title}
    </Headline>
    <Image src={article.feature_image} />
    <Box
      align="center"
      justify="center"
      className={`${styles.articleWrapper} markdown-body`}
      pad={{ horizontal: 'small' }}
    >
      <Markdown
        className={styles.content}
        content={article.content}
        components={{
          h1: { props: { strong: true } },
          h2: { props: { strong: true } },
          p: { props: { size: 'large' } },
          img: { props: { size: 'small' } },
        }}
      />
    </Box>
  </Article>
);

SingleArticleContent.propTypes = {
  article: PropTypes.object.isRequired,
};

export default cssModules(SingleArticleContent, styles);
