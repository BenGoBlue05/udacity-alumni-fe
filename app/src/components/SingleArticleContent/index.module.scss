.singleArticle {
  position: relative;
  width: 100%;
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(46,61,73,.2);
  border: 1px solid #dbe2e8;
  max-width: 100vw;
  box-sizing: border-box;
  padding: 60px 60px !important;
  @media screen and (max-width: 768px) {
    padding: 60px 30px !important;
  }
  @media screen and (max-width: 768px) {
    padding: 20px 10px !important;
  }
}

.articleDate {
  margin-bottom: 40px;
  @media screen and (min-width: 768px) {
    margin-bottom: 0;
    left: 3.0em;
    top: 74px;
    z-index: 2;
    position: absolute;
    transform: translate(-100%, 0);
    margin-left: 0px;
  }
}

.articleWrapper {
  margin-top: 20px;
  p {
    max-width: 640px !important;
    font-weight: 400 !important;
    font-style: normal !important;
    font-size: 21px !important;
    line-height: 1.58 !important;
    letter-spacing: -.003em !important;
  }
}
