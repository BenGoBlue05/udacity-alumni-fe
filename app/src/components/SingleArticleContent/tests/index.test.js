import SingleArticleContent from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<SingleArticleContent />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <SingleArticleContent />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
