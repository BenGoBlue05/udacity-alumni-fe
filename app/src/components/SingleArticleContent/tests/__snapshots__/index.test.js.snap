exports[`<SingleArticleContent /> should render with default props 1`] = `
<Article
  align="center"
  direction="column"
  pad="none">
  <div>
    <undefined
      date="2016-09-30T02:39:57.333Z" />
  </div>
  <Headline
    align="center">
    Creditable coverage Hcfa common procedure coding system Group health plan
  </Headline>
  <undefined />
  <Image
    size="medium"
    src="https://robohash.org/magnietnon.png?size=300x300" />
  <Box
    align="center"
    announce={false}
    className="undefined markdown-body"
    direction="column"
    focusable={true}
    justify="center"
    pad={
      Object {
        "horizontal": "small"
      }
    }
    responsive={true}
    tag="div">
    <Markdown
      components={
        Object {
          "h1": Object {
            "props": Object {
              "strong": true
            }
          },
          "h2": Object {
            "props": Object {
              "strong": true
            }
          },
          "img": Object {
            "props": Object {
              "size": "small"
            }
          },
          "p": Object {
            "props": Object {
              "size": "large"
            }
          }
        }
      }
      content="Care plan eligible dependent free-look period waiting period incurral date board certified health care provider HCPCS. Agent of record PCP limited policy morbidity medigap health savings account credentialing medical necessity open enrollment. Creditable coverage agent of record renewal disenroll IPA medicare. Accumulation period contract year policy benefit cap medicare supplement creditable coverage." />
  </Box>
</Article>
`;
